File: 프로그래머스/lv2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.swift
import Foundation

func solution(_ bridge_length:Int, _ weight:Int, _ truck_weights:[Int]) -> Int {
    var ing = Array(repeating: 0, count: bridge_length)
    var w = 0
    var truck = truck_weights
    var sec = 0
    while !ing.isEmpty {
        sec += 1
        w -= ing.removeFirst()
        
        if let t = truck.first {
            if t + w <= weight {
                ing.append(t)
                w += truck.removeFirst()
            }else {
                ing.append(0)
            }
        }
    }
    return sec
}

File: 프로그래머스/lv2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.py
def solution(bridge_length, weight, truck_weights):
    answer = 0
    ing = [0] * bridge_length
    w = 0
    
    while ing:
        answer += 1
        w -= ing.pop(0)
        
        if truck_weights:
            if w + truck_weights[0] <= weight:
                newTruck = truck_weights.pop(0)
                ing.append(newTruck)
                w += newTruck
            else:
                ing.append(0)
            
        
    return answer